### DISCUSSION TITLE
Decompose with Compose Multiplatform for Web

### DISCUSSION THREAD

**Discussion by devjorgecastro** [arkivanov/Decompose#644] [Q&A] [answered]
Is there any example of how to use Decompose with Kotlin/Wasm using Compose Multiplatform?
I have created a web-only project and I have not been able to configure it with Decompose

**Comment by arkivanov** [comment, marked as answer]
It should be pretty similar to Compose for Web JS, see the sample [here](https://github.com/arkivanov/Decompose/tree/v3.0/sample/app-ios-compose). Also, I recently created the [Minesweeper](https://github.com/arkivanov/Minesweeper) game that supports Compose for Wasm, but the use of Compose is pretty limited there.

**Reply by devjorgecastro** [reply]
It was what I needed, I was able to make an example by reviewing these two repositories, thank you very much 🙂

**Reply by devjorgecastro** [reply]
I created this simple article as an example to get you started with Compose Multiplatform for Web and Decompose:
https://devjorgecastro.medium.com/building-a-simple-web-application-with-compose-multiplatform-using-decompose-d54f2e75631a

**Reply by arkivanov** [reply]
Nice one! Thanks for sharing! I think you are missing `lifecycle.resume()`, see the example [here](https://github.com/arkivanov/Minesweeper/blob/5a01d6fdd2987bea7a2c94a36b7c9fb9fd366d68/composeApp/src/wasmJsMain/kotlin/Main.kt#L28). Without it, none of the lifecycle events will be ever triggered.

You can also attach `LifecycleRegistry` to `document`, as shown [here](https://github.com/arkivanov/Decompose/blob/8d56415e219ccd8d8dbf68d5fe58c7695daa6add/sample/app-js/src/main/kotlin/com/arkivanov/sample/app/Main.kt#L40).

**Reply by devjorgecastro** [reply]
Thanks for the suggestion 🙂, I will update the article.
